api/v1beta1/cluster_types.go

```diff
@@ -332,6 +332,11 @@ type Topology struct {
 	// The name of the ClusterClass object to create the topology.
 	Class string `json:"class"`
 
+	// The namespace of the ClusterClass object to create the topology.
+	//
+	// +optional
+	ClassNamespace string `json:"classNamespace,omitempty"`
```

Comment:
Out of curiosity.
Are we planning to refactor Class/ClassName down the line or keep as it is (top of mind, having name/namespace for an referenced object inlined in the parent struct/not using a ref struct is sort of unique in Cluster API).

cc @vincepri @enxebre  @JoelSpeed who might have opinions on API modeling as well

Reply:
In a future version of this API, I'd expect us to move to
```
class:
  name: ...
  namespace: ...
```
As part of a clean up, this would then be in-keeping, as you say, with other APIs we have

Reply:
@Danil-Grigorev, before this PR merge, could you kindly open an issue with kind api-change with the API design proposed by @JoelSpeed , so we can easily keep it in the radar while working to the v1beta2 API?

Reply:
Added: https://github.com/kubernetes-sigs/cluster-api/issues/11500